interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const multipassSupportSummary = `allows operating as the Multipass service`
+
+const multipassSupportBaseDeclarationPlugs = `
+  multipass-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const multipassSupportBaseDeclarationSlots = `
+  multipass-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const multipassSupportConnectedPlugAppArmor = `
+# Description: allow operating as the Multipass daemon. This policy intentionally
+# allows the Multipass daemon to configure AppArmor, and we expect Multipass to
+# create and apply correct policies for the child processes it spawns.
+
+# Allow socket
+/run/multipass_socket  rw,
+capability chown,
+capability fsetid,
+
+# Multipass generates AppArmor profiles for the child processes it spawns
+# Allow it to add these profiles and run the child under that profile.
+/usr/sbin/aa-exec ixr,
+/sbin/apparmor_parser ixr,
+/etc/apparmor.d/cache/ r,
+/etc/apparmor.d/cache/.features r,
+/etc/apparmor.d/{,cache/}multipass* rw,
+/etc/apparmor.d/tunables/{,**} r,
+/etc/apparmor.d/abstractions/{,**} r,
+/etc/apparmor/parser.conf r,
+/etc/apparmor/subdomain.conf r,
+/sys/kernel/security/apparmor/.replace w,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/{,**} r,
+
+capability net_admin,
+capability mac_admin,
```

Comment:
This is obviously needed since you are loading your own policy. That said, it allows you to modify your own policy. I sort of want to recommend using a child profile, but that doesn't help a lot since you will need to call apparmor_parser to load it and it of course needs this.

We spoke about privilege separation in the multipassd code. Are you doing any of that with regard to loading policy? If so, it would be nice to add a brief comment on that.

Reply:
Yes that's the ultimate goal, privilege separation along with apparmor confinement. I've updated the intro comment to suggest this (is "all possible security meachanisms" too vague?)

I've not got the privilege separation stuff working yet though ([1]). I hope that doesn't delay this review.

[1] https://github.com/CanonicalLtd/multipass/compare/strict-snap-part1?expand=1#diff-87e6e776daf42b26accf0a1d40020cbfR74

Reply:
Comments added as requested